Scc 3853/url and display string

charmingduchess opened the pull request:
- performance work to remove extraneous rerenders and `useEffect` call
  - removing `useEffect` with dependency on `appliedFilters` and instead passing a `submitFilters` function down to filter components.
- create the filter display string (ie "34 Items matched filtered by xyz") based on the query params provided by the next router
- styling display string based on Design System components
- stray bug and styling fixes, not limited to:
  - removing space from the filter bar
  - making sure count on filter bar reflects applied and not just selected filters

EdwinGuzman commented on src/utils/utilsTests/itemFilterUtils.test.tsx:
```diff
@@ -84,55 +84,40 @@ describe("Item Filter Utils", () => {
 
-  describe("buildAppliedFiltersString", () => {
-    const query = {
+  describe.only("buildAppliedFiltersString", () => {
```
Remove the `.only`.

EdwinGuzman reviewed (changes requested):
Works well for Location and Status! For Format, it doesn't seem to have the right id/key for the last two checkboxes and renders `Filtered by format: 'undefined'`
<img width="500" alt="Screen Shot 2023-11-27 at 4 03 52 PM" src="https://github.com/NYPL/research-catalog/assets/1280564/21d566f2-363f-4e97-861d-c87601f7676c">

Also added comments for updated patterns.

charmingduchess commented:
> Works well for Location and Status! For Format, it doesn't seem to have the right id/key for the last two checkboxes and renders `Filtered by format: 'undefined'` <img alt="Screen Shot 2023-11-27 at 4 03 52 PM" width="500" src="https://private-user-images.githubusercontent.com/1280564/286020429-21d566f2-363f-4e97-861d-c87601f7676c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDExODMxOTAsIm5iZiI6MTcwMTE4Mjg5MCwicGF0aCI6Ii8xMjgwNTY0LzI4NjAyMDQyOS0yMWQ1NjZmMi0zNjNmLTRlOTctODYxZC1jODc2MDFmNzY3NmMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTEyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzExMjhUMTQ0ODEwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTVhZGExNzllY2ViOGU2ZmJjNjJlMjdkZDU0M2Y1N2RhNDZjOTgzNWMxNGJkNGM5ZmM0MDg3YTU3YTE1NzZjNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.EKkaOo9Sk1hUAzUSFTaMZQmFZo67XEDd_Gpevbtrrl8">
> 
> Also added comments for updated patterns.
@EdwinGuzman this is a data issue - there are dates in the format section which breaks the internal logic. i updated the fixture on a later branch to use more standard formats. paul said we should not account for bad data.

EdwinGuzman commented:
> @EdwinGuzman this is a data issue - there are dates in the format section which breaks the internal logic. i updated the fixture on a later branch to use more standard formats. paul said we should not account for bad data.

Great, sounds good.

charmingduchess commented on src/components/ItemFilters/FiltersContainer.tsx:
```diff
@@ -17,62 +24,76 @@ interface ItemFilterContainerProps {
-  }, [query])
-
-  useEffect(tempSubmitFilters, [activeFilters, tempSubmitFilters])
+  useMemo(() => {
```
awesome! yes this makes sense.

EdwinGuzman commented on src/components/ItemFilters/FiltersContainer.tsx:
```diff
@@ -17,62 +23,75 @@ interface ItemFilterContainerProps {
-  })
+  const router = useRouter()
+
+  const filterData = useRef(
```
For type safety and to better see how `filterData` is structured, can you type this as `useRef<ItemFilterData[]>(`

EdwinGuzman commented on src/components/ItemFilters/FiltersContainer.tsx:
```diff
@@ -17,62 +23,75 @@ interface ItemFilterContainerProps {
   const [whichFilterIsOpen, setWhichFilterIsOpen] = useState("")
 
-  const [tempQueryDisplay, setTempQueryDisplay] = useState("")
+  const itemsMatched = buildItemsString(router.query)
```
What do you think about renaming the function as `buildItemsMatchedString` or is that too long?

EdwinGuzman commented on src/utils/itemFilterUtils.ts:
```diff
@@ -51,27 +51,30 @@ export const buildItemFilterQueryParams = (
-  numItems = 20,
-  itemAggs: ItemFilterData[]
-) => {
+export const buildItemsString = (query, numItems = 20) => {
```
Why is 20 the default?

charmingduchess replied:
it's just for temporary display purposes because i don't know what kind of data is going to be passed in from the bib page. ill add a comment.

EdwinGuzman reviewed (approved):
Minor comments but the update looks good.
